### PR TITLE
Script to generate Doxygen and Create a Zip file!

### PR DESCRIPTION
**Pull request by sarenameas**
- Generate doyxgen and optionally generate a zip file.
  ```
  python3 docs/doxygen/generate_docs.py --root <csdk_root> --zip
  ```

- Folder structure of the html output is preserved relative to the root of the CSDK. This is to properly use the referenced TAG files in the spoke repos. The references are relative paths from the hub html output.

- Please see the generated zip:
[doxygen.zip](https://github.com/aws/aws-iot-device-sdk-embedded-C/files/5211386/doxygen.zip)

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: docs/doxygen/generate_docs.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
```
Comment (muneebahmed10) [inline docs/doxygen/generate_docs.py:1]: Should this file be in `tools/` instead of `docs/`?

File: docs/doxygen/generate_docs.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        )
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        result = e.stdout
+        return result
+
+
+def get_lib_paths(root):
+    """
+    Get all of the paths, relative to the root, of to the libraries under the
```
Comment (muneebahmed10) [inline docs/doxygen/generate_docs.py:32]:
```suggestion
    Get all of the paths, relative to the root, of the libraries under the
```

File: docs/doxygen/generate_docs.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    """
+    standard_libs_path = os.path.join(root, "libraries", "standard")
+    aws_libs_path = os.path.join(root, "libraries", "aws")
+    abs_lib_paths = []
+
+    lib_dirs = os.listdir(standard_libs_path)
+    abs_lib_paths += list(map(lambda lib_dir: os.path.join(standard_libs_path, lib_dir), lib_dirs))
+    lib_dirs = os.listdir(aws_libs_path)
+    abs_lib_paths += list(map(lambda lib_dir: os.path.join(aws_libs_path, lib_dir), lib_dirs))
+    return abs_lib_paths
```
Comment (muneebahmed10) [inline docs/doxygen/generate_docs.py:43]: How about:
```suggestion
    libs_classifications = [ "standard", "aws" ]
    abs_lib_paths = []

    for lib in libs_classifications:
        libs_path = os.path.join(root, "libraries", lib)
        lib_dirs = os.listdir(libs_path)
        abs_lib_paths += list(map(lambda lib_dir: os.path.join(libs_path, lib_dir), lib_dirs))
    return abs_lib_paths
```

File: README.md [hunk 4]
```diff
@@ -174,11 +174,11 @@ docker run -p 80:80 kennethreitz/httpbin
...
 
 ```shell
-cd libraries/standard/coreMQTT
-doxygen docs/doxygen/config.doxyfile
-cd ../../..
-doxygen docs/doxygen/config.doxyfile
-```
\ No newline at end of file
+cd <CSDK_ROOT>
+python3 docs/doxygen/generate_docs.py --root .
```
Comment (yourslab) [inline README.md:181]: Should submodules be cloned before calling the python script?

File: docs/doxygen/generate_docs.py (hunk 1 above)
Comment (sarenameas) [reply docs/doxygen/generate_docs.py:1]: Was pondering this last night.

File: README.md (hunk 4 above)
Comment (sarenameas) [reply README.md:181]: Yes. But I didn't want to make it part of the script.

File: docs/doxygen/generate_docs.py (hunk 3 above)
Comment (sarenameas) [reply docs/doxygen/generate_docs.py:43]: sure

File: docs/doxygen/generate_docs.py (hunk 2 above)
Comment (sarenameas) [reply docs/doxygen/generate_docs.py:32]: thanks

**Review by muneebahmed10 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**